sdk/lib/rtl/heap.c

```diff
@@ -2359,10 +2417,7 @@ RtlpGrowBlockInPlace (IN PHEAP Heap,
     /* Get entry flags */
     EntryFlags = InUseEntry->Flags;
 
-    /* Get the next free entry */
-    FreeEntry = (PHEAP_FREE_ENTRY)(InUseEntry + InUseEntry->Size);
-
```

Comment:
Because
`(PVOID)PAGE_ROUND_UP(InUseEntry + InUseEntry->Size))` 
is different from
`InUseEntry + InUseEntry->Size`
There can be a guard entry behind InUseEntry (which RtlpIsLastCommittedEntry checks), and thus `InUseEntry + InUseEntry->Size` is no longer the beginning of an uncommitted range, but the end of the page it resides on is.